cmd/bash/main.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Base image expects shell in $PATH
+shell command is used for creating directory with sub-command `mkdir`,
+copy contents between directories with sub-command `cp`.
+*/
```

Comment:
Sorry for continuing to harp on this, but I still don't understand what this image is _used_ for - i.e. when would I use the `bash-noop-image` image, and why do we need to be building and publishing this image?

Reply:
> To publish knative release with open source licensed compliant base image instead of busybox(license nightmare!).

oohhhhhhh it all makes sense now! :D - maybe mention this in the main.go or similar location?

Reply:
> To publish knative release with open source licensed compliant base image instead of busybox(license nightmare!).

Okay i'm still a bit confused but I think I'm almost there.

So this link in `ko.yaml`:
`github.com/knative/build-pipeline/cmd/bash: busybox # image should have shell in $PATH`

Looking at [the ko docs](https://github.com/google/go-containerregistry/tree/master/cmd/ko#overriding-the-default-base-image) doesn't this line say that we _will_ be using `busybox` as the base image when building `github.com/knative/build-pipeline/cmd/bash`, i.e. that `github.com/knative/build-pipeline/cmd/bash` will using `busybox` as a base image? And so would we still have the license issue?

> To publish knative release with open source licensed compliant base image instead of busybox(license nightmare!).

Second question: which images do we need to build on top of this image? Is the gcs util image meant to be built on top of the bash image?

Reply:
> And so would we still have the license issue?

knative release does not have to be license compliant(I think).
If vendors want to release knative with different base image then they can override `.ko.yaml` with their licensed base image. 
 
> Second question: which images do we need to build on top of this image? Is the gcs util image meant to be built on top of the bash image?

No. `gcs-util` image is build on top of `google/cloud-sdk:alpine` image.